File: WebsiteAppRPG/Core/Entities/MapRouter.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace WebsiteAppRPG.Core.Entities
{
    [Table("MapRouters")]
    public class MapRouter
    {
        [Key]
        [Column("MapRouterID")]
        [JsonPropertyName("mapRouterId")]
        public int MapRouterID { get; set; }

        [ForeignKey("MapID")]
        [Column("MapID")]
        [JsonPropertyName("mapId")]
        public int MapID { get; set; }

        [Column("EnterPositionX")]
        public int EnterPositionX { get; set; }

        [Column("EnterPositionY")]
        public int EnterPositionY { get; set; }

        [Column("ExitPositionX")]
        public int ExitPositionX { get; set; }

        [Column("ExitPositionY")]
        public int ExitPositionY { get; set; }

        [Column("DestinationMapID")]
        [JsonPropertyName("destinationMapId")]
        public int DestinationMapID { get; set; }

        public MapRouter()
        {
            
        }

        public MapRouter(int mapId, int enterPositionX, int enterPositionY, int exitPositionX, int exitPositionY)
        {
            MapID = mapId;
            EnterPositionX = enterPositionX;
            EnterPositionY = enterPositionY;
            ExitPositionX = exitPositionX;
            ExitPositionY = exitPositionY;
        }

    }
}

File: WebsiteAppRPG/Frontend/website-app-rpg-frontend/src/models/mapRouter.ts
export class MapRouter{

    public mapRouterId: number;

    public mapId: number;

    public enterPositionX: number;

    public enterPositionY: number;

    public exitPositionX: number;

    public exitPositionY: number;

    public destinationMapId :number;

    constructor(mapRouterId: number, mapId: number, enX: number, enY: number, exX: number, exY: number, destinationMapId: number) {
        this.mapRouterId = mapRouterId;
        this.mapId = mapId;
        this.enterPositionX = enX;
        this.enterPositionY = enY;
        this.exitPositionX = exX;
        this.exitPositionY = exY;
        this.destinationMapId = destinationMapId;
    }
}
